sky/backends/backend_utils.py

```diff
@@ -2014,17 +2014,24 @@ def run_ray_status_to_check_ray_cluster_healthy() -> bool:
             # triggered.
             if external_ips is None or len(external_ips) == 0:
                 logger.debug(f'Refreshing status ({cluster_name!r}): No cached '
-                             f'IPs found. External IPs: {external_ips}')
+                             f'IPs found. Handle: {handle}')
                 raise exceptions.FetchIPError(
                     reason=exceptions.FetchIPError.Reason.HEAD)
 
+            if handle.head_ssh_port is None:
+                # Refresh the ssh ports. It is ok to refresh as it is fast.
+                handle.external_ssh_ports()
+                if handle.head_ssh_port is None:
+                    logger.debug(
+                        f'Refreshing status ({cluster_name!r}): failed '
+                        f'to get the ssh ports. Handle: {handle}')
+                    raise exceptions.FetchIPError(
```

Comment:
Shall we use a new exception named `FetchPortError` instead?

Reply:
Changed to `RuntimeError`. Wdyt?

Reply:
Sounds good!